Azure/azure-cli
# [ARM] `az deployment group/sub/mg/tenant create`: Add `--what-if` and `--proceed-if-no-change` parameters

**Description**
Closes #18481.
Closes https://github.com/Azure/bicep/issues/2923

**Testing Guide**
```
az deployment sub -l westus -f ./deploy.json --what-if
az deployment sub -l westus -f ./deploy.json --confirm-with-what-if --proceed-if-no-change
```

**History Notes**


[ARM] az deployment group/sub/mg/tenant create: Add --what-if parameter for invoking What-If with the deployment create commands.
[ARM] az deployment group/sub/mg/tenant create: Add --proceed-if-no-change parameter to skip confirmation when --confirm-with-what-if is set and there's no changes in What-If results.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
